File: 2/fibs.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func helpMe(errmsg string) int {
	fmt.Println(errmsg)
	fmt.Println()
	s := fmt.Sprintf("Usage:\n\t%s <integer>\n", os.Args[0])
	fmt.Println(s)
	fmt.Println("Calculate sum of even fibonacci values up until")
	fmt.Println("provided max value.")
	return 5
}

func fibonext(a int, b int) (int, int) {
	// fmt.Println("Input: ", a, b) // DEBUG
	a += b
	b += a
	// fmt.Println("Output: ", a, b) // DEBUG
	return a, b
}

func main() {
	if len(os.Args) != 2 {
		os.Exit(helpMe("ERR This takes a single integer as an argument."))
	}
	
	// convert the argument to an integer
	max, _ := strconv.Atoi(os.Args[1])
	value := 0

	// initialize the first two fibonacci values
	x := 1
	y := 2

	for y < max {
		if x%2 == 0 {
			value += x
		}
		
		if y%2 == 0 {
			value += y
		}

		x, y = fibonext(x, y)	
	}

	fmt.Println(value)
}

File: 2/fibs.py
#!/usr/bin/env python3
#-*- coding: utf-8 -*-

import sys

'''
Calculate the Fibonacci sequence until provided value.
Take even values in the sequence between 1 and that value
and sum only the even values.
'''

def fibonacci_get_next(x,y):
    '''calculate the next two values.'''
    nx = x + y
    ny = y + nx
    return nx,ny

def print_help(msg=None):
    '''Pretty help in a lazy-argument-handling world.'''
    if msg:
        print(msg)
 
    print("Usage:")
    print("\t{} integer [debug]".format(sys.argv[0]))
    print("\nSum even values in the Fibonacci sequence"+
             " from 1 to provided upper bound.")

def main():
    # if they want help, do nothing else.
    if 'help' in sys.argv:
        print_help()
        return 1

    x = 1
    y = 2
    debug = False
    evens = []

    # lazy command line argument handling,
    while 'debug' in sys.argv:
        sys.argv.remove('debug')
        debug = True

    if len(sys.argv) == 1:
        print_help("ERR You need to provide an integer to use as" + 
                   " the upper bound!")
        return 5

    # main logic.
    while y < int(sys.argv[1]):
        for i in x,y:
            if i % 2 == 0:
                evens.append(i)
        x, y = fibonacci_get_next(x,y) 

    # provide output.
    if debug:
        print('DEBUG calculated the following list '+
              'of even values: {}'.format(evens))

    print('Result: {}'.format(sum(evens)))

if __name__ == '__main__':
    sys.exit(main())
